TestFoundation/TestURLCache.swift

```diff
@@ -0,0 +1,271 @@
+        }
+        
+        try FileManager.default.removeItem(at: writableTestDirectoryURL)
+        
+        do {
+            let cache = try self.cache(memoryCapacity: lots, diskCapacity: lots)
+            
+            let (request, response) = try cachePair(for: "https://google.com/", ofSize: aBit, storagePolicy: .notAllowed)
+            cache.storeCachedResponse(response, for: request)
+            
+            XCTAssertEqual(try FileManager.default.contentsOfDirectory(atPath: writableTestDirectoryURL.path).count, 0)
+            XCTAssertNil(cache.cachedResponse(for: request))
+        }
+        
+        try FileManager.default.removeItem(at: writableTestDirectoryURL)
+    }
+    
+    func testNoDiskUsageIfDisabled() throws {
+        let cache = try self.cache(memoryCapacity: lots, diskCapacity: 0)
+        let (request, response) = try cachePair(for: "https://google.com/", ofSize: aBit)
+        cache.storeCachedResponse(response, for: request)
+        
+        XCTAssertEqual(try FileManager.default.contentsOfDirectory(atPath: writableTestDirectoryURL.path).count, 0)
+        XCTAssertNotNil(cache.cachedResponse(for: request))
+    }
+    
+    func testShrinkingDiskCapacityEvictsItems() throws {
+        let cache = try self.cache(memoryCapacity: lots, diskCapacity: lots)
+        
+        let urls = [ "https://apple.com/",
+                     "https://google.com/",
+                     "https://facebook.com/" ]
+        
+        for (request, response) in try urls.map({ try cachePair(for: $0, ofSize: aBit) }) {
+            cache.storeCachedResponse(response, for: request)
+        }
+        
+        XCTAssertEqual(try FileManager.default.contentsOfDirectory(atPath: writableTestDirectoryURL.path).count, 3)
+        for url in urls {
+            XCTAssertNotNil(cache.cachedResponse(for: URLRequest(url: URL(string: url)!)))
+        }
+        
+        cache.diskCapacity = 0
+        XCTAssertEqual(try FileManager.default.contentsOfDirectory(atPath: writableTestDirectoryURL.path).count, 0)
+        for url in urls {
+            XCTAssertNotNil(cache.cachedResponse(for: URLRequest(url: URL(string: url)!)))
+        }
+    }
+    
+    func testNoMemoryUsageIfDisabled() throws {
+        let cache = try self.cache(memoryCapacity: 0, diskCapacity: lots)
+        let (request, response) = try cachePair(for: "https://google.com/", ofSize: aBit)
+        cache.storeCachedResponse(response, for: request)
+        
+        XCTAssertEqual(try FileManager.default.contentsOfDirectory(atPath: writableTestDirectoryURL.path).count, 1)
+        XCTAssertNotNil(cache.cachedResponse(for: request))
+        
+        // Ensure that the fulfillment doesn't come from memory:
+        try FileManager.default.removeItem(at: writableTestDirectoryURL)
+        try FileManager.default.createDirectory(at: writableTestDirectoryURL, withIntermediateDirectories: true)
+        
+        XCTAssertNil(cache.cachedResponse(for: request))
```

Comment:
@drodriguez The test here is explicitly about checking that the one `URLCache` object did use disk but didn't use memory if configured with `memoryCapacity: 0`.